Sources/MapboxNavigation/WayNameView.swift

```diff
@@ -2,12 +2,161 @@ import Foundation
+        self.representation = imageRepresentation
+        setUpWith(roadName: roadName)
+    }
+    
+    // Set up the `WayNameLabel` with the road name. Try to use the Mapbox designed shield first, if failed, fall back to use the legacy road shield icon.
+    // If there's no valid shield image, display the road name only.
+    private func setUpWith(roadName: String) {
+        var didSetUp = false
+        
+        if let shield = representation?.shield,
+           let shieldIcon = spriteRepository.getShield(displayRef: shield.text, name: shield.name) {
+            didSetUp = setupWithShield(roadName: roadName, shieldIcon: shieldIcon)
+        }
+        
+        if !didSetUp,
+           let shieldLegacyIcon = spriteRepository.getLegacyShield() {
+            didSetUp = setupWithLegacy(roadName: roadName, shieldIcon: shieldLegacyIcon)
+        }
+        
+        if !didSetUp {
+            text = roadName
+        }
+    }
+
+    /**
+     Fills contents of the `WayNameLabel` with the road name and legacy shield icon.
+     
+     - parameter roadName: The road name `String` that should be presented on the view.
+     - parameter shieldIcon: The legacy `UIImage` that represents the current road.
+     - returns: `true` if operation was successful, `false` otherwise.
+     */
+    @discardableResult
+    private func setupWithLegacy(roadName: String, shieldIcon: UIImage) -> Bool {
+        var currentShieldName: NSAttributedString?, currentRoadName: String?
+        var didSetup = false
+
+        let attachment = ShieldAttachment()
+        let fontSize = frame.size.height / 2.5
+        attachment.image = shieldIcon.withFontSize(font: UIFont.boldSystemFont(ofSize: fontSize),
+                                                   size: frame.size)
+        currentShieldName = NSAttributedString(attachment: attachment)
+
+        if !roadName.isEmpty {
+            currentRoadName = roadName
+            text = roadName
+            didSetup = true
+        }
+
+        if let compositeShieldImage = currentShieldName, let roadName = currentRoadName {
+            let compositeShield = NSMutableAttributedString(string: " \(roadName)")
+            compositeShield.insert(compositeShieldImage, at: 0)
+            attributedText = compositeShield
+            didSetup = true
+        }
+        return didSetup
+    }
+    
+    /**
+     Fills contents of the `WayNameLabel` with the road name and shield icon from the spriteRepository.
+     
+     - parameter roadName: The road name `String` that should be presented on the view.
+     - parameter shieldIcon: The  `UIImage` that represents the current road shield.
+     - returns: `true` if operation was successful, `false` otherwise.
+     */
+    @discardableResult
+    private func setupWithShield(roadName: String, shieldIcon: UIImage) -> Bool {
+        guard let shield = representation?.shield else { return false }
```

Comment:
The class store a copy of it because when the shield `ImageRepresentation` changes, we need to update the cache to re-download the sprite sheet, sprite info and the legacy shield icon. We only update the cache when the  `ImageRepresentation` change or when the map style changes.